src/tsolvers/egraph/EgraphSolver.cc

```diff
@@ -330,20 +332,24 @@ void Egraph::constructTerm(PTRef tr) {
 
         boolTermToERef.insert(tr_pure, er_pure);
         boolTermToERef.insert(tr_neg, er_neg);
-    }
-
-    else {
-        sym = enode_store.addSymb(tm.symb());
-        cdr = ERef_Nil;
-        for (int j = tm.size() - 1; j >= 0; j--) {
-            assert(enode_store.termToERef.has(tm[j])); // The child was not inserted
-            ERef car = enode_store.termToERef[tm[j]];
+    } else {
+
+        if (not logic.isNonBoolIte(tr)) {
```

Comment:
Also here, it seems like boolean and non-boolean ITE should be treated the same.